src/hyperion/device_setup_plans/dcm_pitch_roll_mirror_adjuster.py

```diff
@@ -0,0 +1,167 @@
+import bluesky.plan_stubs as bps
+from dodal.devices.DCM import DCM
+from dodal.devices.hfm import HFM
+from dodal.devices.i0 import I0
+from dodal.devices.qbpm1 import QBPM1
+from dodal.devices.s4_slit_gaps import S4SlitGaps
+from dodal.devices.vfm import VFM
+from ophyd import EpicsMotor, EpicsSignalRO
+
+from hyperion.device_setup_plans.peak_finder import (
+    PeakFinder,
+    SimpleMaximumPeakEstimator,
+    SingleScanPassPeakFinder,
+)
+from hyperion.log import LOGGER
+from hyperion.utils.lookup_table import LinearInterpolationLUTConverter
+
+RESET_SLIT_GAP_GROUP = "reset_slit_gap"
+
+PREP_FOR_VFM_GROUP = "prepare_for_vfm_align"
+PREP_FOR_HFM_GROUP = "prepare_for_hfm_align"
+
+# Magic values from GDA beamLineSpecificEnergy.py
+SLIT_SIZE_WIDE_OPEN_MM = (500, 500)
+SLIT_SIZE_LARGE_MM = (500, 60)
+SLIT_SIZE_SMALL_MM = (100, 60)
+
+VFM_FINE_PITCH_INITIAL_PRESET_MM = 0.019
+VFM_FINE_PITCH_ADJUST_PEAK_WIDTH_MM = 0.012
+VFM_FINE_PITCH_ADJUST_STEP_MM = 0.0006
+
+HFM_FINE_PITCH_INITIAL_PRESET_MM = 0.017
+HFM_FINE_PITCH_ADJUST_PEAK_WIDTH_MM = 0.016
+HFM_FINE_PITCH_ADJUST_STEP_MM = 0.0006
+
+DCM_PITCH_ADJUST_PEAK_WIDTH = 0.075
+DCM_PITCH_ADJUST_PEAK_STEP = 0.002
+
+
+class DCMPitchRollMirrorAdjuster:
```

Comment:
Should: We tend not to hold plans in a class like this, instead choosing to have a composite with the devices and bare generators for the plans (take a look at the other plans). We want to do this as:
* It integrates better with the way `blueapi` (the API package we want to move to) works
* General consistency

Reply:
I can move the detectors into a composite but I don't think the peak finder and estimator should be included as they aren't really value objects. The purpose of the class is really to separate the domain function (configure the optics) from the strategy (how to find peaks). I'm expecting that it would be invoked by a wrapper function that would instantiate whatever implementations of finders/estimators from a serialized form, that would presumably have some kind of composite. But I haven't written it as I don't know what the ultimate shape of that would be, whether it would consist of only this section of code or whether it would consist of other plan elements as well.